### PR TITLE
[Feature] Wildcard Indexing

### PR DESCRIPTION
**Pull request by JonoPrest**
I will use this as a feature branch for wildcard indexing. The rest of my PRs will be based on this and I will merge when the feature is complete or in a state that is non breaking

### CODE REVIEW

**Comment by JonoPrest** [comment]
First PR: 
- adds LogSelection module as a type that can translate into both hypersync and rpc queries (in future)

**Comment by DZakh** [comment]
I'll review it tomorrow

File: codegenerator/cli/templates/static/codegen/src/LogSelection.res [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+let topicSelectionHasFilters = (topicSelection: topicSelection) =>
+  switch topicSelection {
+  | {topic1: [], topic2: [], topic3: []} => false
+  | _ => true
+  }
+
+let hasTopicFilters = ({topicSelections}: t) =>
+  topicSelections->Belt.Array.reduce(false, (accum, item) => {
+    accum || item->topicSelectionHasFilters
+  })
```
Comment (DZakh) [inline codegenerator/cli/templates/static/codegen/src/LogSelection.res:40]: It's better to use `Array.some` for cases like this. It's more readable and will stop iterating on the first callback returning true.

**Review by DZakh — APPROVED** [review]
I think you can merge it to master immediately, without stacking multiple parts of the feature. It'll help with merge conflicts when I start working on HyperFuel next week

File: codegenerator/cli/templates/static/codegen/src/LogSelection.res (hunk 1 above)
Comment (JonoPrest) [reply codegenerator/cli/templates/static/codegen/src/LogSelection.res:40]: Nice I didn't know that existed 🙂

**Pull request merged**